### PR TITLE
vim-patch:partial:016188fd8a30

### PR DESCRIPTION
**Pull request by clason**
Update runtime files.
https://github.com/vim/vim/commit/016188fd8a30cfbaca3faa0daea9a47138dc5c4b

skip manual rewrite for Vim9script

### CODE REVIEW

File: runtime/doc/builtin.txt [hunk 1]
```diff
@@ -3170,7 +3209,8 @@ getmatches([{win}])					*getmatches()*
 		as |setmatches()| can restore a list of matches saved by
 		|getmatches()|.
 		If {win} is specified, use the window with this number or
-		window ID instead of the current window.
+		window ID instead of the current window.  If {win} is invalid,
+		an empty list is returned.
```
Comment (seandewar) [inline runtime/doc/builtin.txt:3213]: actually seems to return 0 for nvim, probably a minor inaccuracy with the port of v8.1.1084

**Review by seandewar — COMMENTED** [review]
missing char2nr

also, maybe they forgot to mention that `acos` returns `0.0` if given an invalid type? (I'd be surprised if it doesn't)

File: runtime/doc/builtin.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/builtin.txt:3213]: _and_ an error, for good measure 😄

**Comment by seandewar** [comment]
> I deliberately omitted all mentions of |String| since I couldn't find that in our builtin.txt

Might be misunderstanding, but `|String|` is referenced in our builtin.txt too (the relevant section is in eval.txt)

> Like :call acos('foo')? That gives an error.

`:echo acos('bar')` as `:call` discards the result

**Comment by clason** [comment]
> > I deliberately omitted all mentions of |String| since I couldn't find that in our builtin.txt
> 
> Might be misunderstanding, but `|String|` is referenced in our builtin.txt too (the relevant section is in eval.txt)

Sounds like it :) I added it.

> > Like :call acos('foo')? That gives an error.
> 
> `:echo acos('bar')` as `:call` discards the result

Yeah, I was confused by the fact that it returns a number _and_ an error, which seems... weird.

But indeed it behaves exactly like `asin`, so I yoinked _that_ doc update instead of the inconsistent one in the upstream patch.

**Review by seandewar — APPROVED** [review]
LGTM.

> Yeah, I was confused by the fact that it returns a number and an error, which seems... weird.

Gotta love Vim script :heart_eyes: 

_Minor nit:_ I dislike the mentions of "nan" (with the quotation marks). At least to me, the quotes imply a returned string, not a float. I think they'd be better written as NaN. :shrug: 

(Not to mention that, stringifying NaN produces "str2float('nan')", not "nan" like in Vim, so you definitely don't want to check for NaNs using strings when `isnan()` exists)

**Comment by seandewar** [comment]
Oh yeah, might want to mark this partial too (e.g: charclass() is ported yet)

**Pull request merged**